torchao/quantization/quant_api.py

```diff
@@ -629,6 +630,33 @@ def int8_dynamic_activation_int4_weight(
     )
 
 
+def gemlite_uintx_weight_only(
+    group_size: Optional[int] = 64,
+    bit_width: int = 4,
+    packing_bitwidth: int = 32,
+    contiguous: Optional[bool] = None,
+):
+    """
+    applies weight only 4 or 8 bit integer quantization and utilizes the gemlite triton kernel and its associated weight packing format.
+    This only works for fp16 models. 8 bit quantization is symmetric, 4 bit quantization is asymmetric.
+
+    Args:
+        `group_size`: parameter for quantization, controls the granularity of quantization, smaller
+         size is more fine grained
+        `bit_width`: bit width of the quantized weight.
+        `packing_bitwidth`: bit width of the packed weight, should be 8 or 32. Can have performance impacts depending on hardware.
+        `contiguous`: if set, the weight will be packed as specified. Leaving it as None lets gemlite determine the best choice.
+    """
+
+    from torchao.dtypes.uintx.gemlite_layout import apply_gemlite_quant
+
+    use_hqq = True if bit_width == 4 else False
+    apply_fn = lambda weight: apply_gemlite_quant(
```

Comment:
Should we raise if gemlite is not installed?

Reply:
I think so, please feel free to review more, and we should have follow up fixes. 

just want to merge now to unblock benchmarking in sglang for now